pontoon/sync/changeset.py

```diff
@@ -112,10 +118,12 @@ def execute_update_vcs(self):
         ).asymmetric().values_list('path', flat=True).distinct()
 
         for path in obsolete_entities_paths:
-            changed_resources.add(resources[path])
+            # Path can't be available if e.g. resource has been removed from the repository.
+            if path in resources:
+                changed_resources.add(resources[path])
 
         if len(obsolete_entities_paths) > 0:
-            self.locales_to_commit = self.locales.values()
+            self.locales_to_commit = set(self.locales.values())
```

Comment:
Nice catch! I guess `self.locales.values()` doesn't contain any duplicates, but it's still nice to keep the original type.